device/pcie/pci_device.cpp

```diff
@@ -23,6 +23,38 @@
 #include "common/assert.hpp"
 #include "common/logger.hpp"
 
+static PciDeviceInfo read_device_info(int fd)
```

Comment:
There is a similar function in get_pcie_device_info and get_pcie_info in utils.hpp, can you just merge this with that, we definitely don't need two ways to call TENSTORRENT_IOCTL_GET_DEVICE_INFO?
Also, I don't like the implementation being in .hpp file. And all the utils in utils.hpp are pci related. Maybe we could just move all of that inside pci_device as private methods (which would make the merge easier then)?

Reply:
I noticed utils.hpp after writing this code.  I will consolidate it into the PCIDevice class.

Reply:
#147 - will take care of that here